var/spack/repos/builtin/packages/relion/package.py

```diff
@@ -58,9 +74,14 @@ def cmake_args(self):
             '-DDoublePrec_CPU=%s' % ('+double' in self.spec),
             '-DDoublePrec_GPU=%s' % ('+double-gpu' in self.spec),
         ]
+        # added cuda arch for p100s
         if '+cuda' in self.spec:
             args += [
                 '-DCUDA=on',
                 '-DCUFFT=on',
             ]
+        if 'cuda_arch' is not None:
```

Comment:
that's gonna give you `true` always. just put 
```
carch = spec.variants['cuda_arch'].value
```
above `if` and do
```
if carch is not None:
   ... // adjust the second line as well
```

Reply:
My bad...